From 8a64ec227477727c4fbe1b19f1a6b099ea766b60 Mon Sep 17 00:00:00 2001
From: Marcin Mielczarczyk <marcin.mielczarczyk@imgtec.com>
Date: Fri, 5 Dec 2014 10:26:10 +0100
Subject: [PATCH 2/2] Felix: Add Felix module to Android

Adds Felix.ko module to Android build system and loads automatically during
Android boot.
---
 device.mk   |    4 ++++
 init.x86.rc |    6 ++++++
 2 files changed, 10 insertions(+)

diff --git a/device.mk b/device.mk
index 44c316a..fd15666 100644
--- a/device.mk
+++ b/device.mk
@@ -14,6 +14,9 @@
 # limitations under the License.
 #
 
+# NOTE: edit this path to point proper Felix.ko location
+FELIX_MODULE_PATH := /home/likewise-open/WR/marcin.mielczarczyk/perforce_ws/felix/DEV/marcin.mielczarczyk_GUC/DDKSource/obj/CI/felix/felix_lib/km/Felix.ko
+
 PRODUCT_DIR := $(dir $(lastword $(filter-out device/common/%,$(filter device/%,$(ALL_PRODUCTS)))))
 
 PRODUCT_PROPERTY_OVERRIDES := \
@@ -37,6 +40,7 @@ PRODUCT_COPY_FILES := \
     $(if $(wildcard $(PRODUCT_DIR)ueventd.$(TARGET_PRODUCT).rc),$(PRODUCT_DIR)ueventd.$(TARGET_PRODUCT).rc,$(LOCAL_PATH)/ueventd.x86.rc):root/ueventd.$(TARGET_PRODUCT).rc \
 
 PRODUCT_COPY_FILES += \
+	$(FELIX_MODULE_PATH):system/lib/modules/Felix.ko \
     $(LOCAL_PATH)/ppp/ip-up:system/etc/ppp/ip-up \
     $(LOCAL_PATH)/ppp/ip-down:system/etc/ppp/ip-down \
     $(LOCAL_PATH)/ppp/peers/gprs:system/etc/ppp/peers/gprs \
diff --git a/init.x86.rc b/init.x86.rc
index 311f98e..9d59670 100644
--- a/init.x86.rc
+++ b/init.x86.rc
@@ -90,6 +90,13 @@ on fs
     setprop ro.crypto.fuse_sdcard true
     # manually start class late_start since we don't use mount_all
     class_start late_start
+	chmod 600 /system/lib/modules/Felix.ko
+	insmod /system/lib/modules/Felix.ko
+	chown media media /dev/imgfelix0
+	chown media media /dev/imgfelixDG0
+	# change owner and group of Felix's I2C PCI device
+	chown media media /sys/bus/pci/devices/0000:01:00.0/resource1
+	chown media media /dev/i2c-0
 
 service wpa_supplicant /system/bin/wpa_supplicant -c/data/misc/wifi/wpa_supplicant.conf \
     -O/data/misc/wifi/sockets \
-- 
1.7.9.5

